include/envoy/network/io_handle.h

```diff
@@ -1,33 +1,136 @@
 #pragma once
 
+#include <bitset>
 #include <memory>
 
 #include "envoy/common/pure.h"
 
 namespace Envoy {
+
+namespace Buffer {
+struct RawSlice;
+}
+
 namespace Network {
 
+namespace Address {
+class Instance;
+} // namespace Address
+
+class IoError {
+public:
+  enum IoErrorCode {
+    IO_NO_ERROR = 0,
+    IO_EAGAIN,
+    IO_ENOTSUP,
+    IO_EAFNOSUPPORT,
+    IO_EINPROGRESS,
+    IO_EPERM,
+    IO_UNKNOWN_ERROR
```

Comment:
Looks great; use `enum class`, lose the IO_ prefix, and use camel-case, e.g. `NoError`.

Reply:
done